tests/asynch/pct.rs

```diff
@@ -0,0 +1,36 @@
+use std::sync::atomic::{AtomicUsize, Ordering};
+
+use shuttle::scheduler::PctScheduler;
+use shuttle::sync::Arc;
+use shuttle::{asynch, Config, MaxSteps, Runner};
+
+// Check that PCT correctly deprioritizes a yielding task. If it wasn't, there would be some
+// iteration of this test where the yielding task has the highest priority and so the others
+// never make progress.
+#[test]
+fn yield_fair() {
+    const NUM_TASKS: usize = 4;
+
+    let scheduler = PctScheduler::new(1, 100);
+    let mut config = Config::new();
+    config.max_steps = MaxSteps::FailAfter(50);
+    let runner = Runner::new(scheduler, config);
+    runner.run(|| {
+        let count = Arc::new(AtomicUsize::new(0usize));
+
+        let _thds = (0..NUM_TASKS)
+            .map(|_| {
+                let count = count.clone();
+                asynch::spawn(async move {
+                    count.fetch_add(1, Ordering::SeqCst);
+                })
+            })
+            .collect::<Vec<_>>();
+
+        asynch::block_on(async move {
+            while count.load(Ordering::SeqCst) < NUM_TASKS {
+                asynch::yield_now().await;
```

Comment:
How about adding a variant of this test without the yield, showing that PCT finds the schedule where the test panics because we hit the max_steps limit?